lib/tether/ops_linux.go

```diff
@@ -183,7 +191,10 @@ func (t *BaseOperations) SetHostname(hostname string, aliases ...string) error {
 	for _, a := range append(aliases, hostname) {
 		Sys.Hosts.SetHost(a, lo4)
 	}
-	if err = Sys.Hosts.Save(); err != nil {
+	if err = Sys.Hosts.Save(hostsPathBindSrc); err != nil {
```

Comment:
So instead of this, you could just set `Sys.Hosts` to `etcconf.NewHosts(hostsPathBindSrc)` in an `init()` function in `/lib/tether/tether.go`. Same goes for `Sys.ResolvConf`. That way you do not need to modify the `Conf` interface.